src/backend/executor/nodeMaterial.c

```diff
@@ -759,3 +764,30 @@ ExecEagerFreeMaterial(MaterialState *node)
 	}
 }
 
+
+/*
+ * mkTempFileForWriter
+ * 
+ * Create a unique tmp file for writer. Then use flock's unblock way to lock the tmp file.
+ * We can make sure the tmp file will be locked forerver until the writer process has quit.
+ */
+void mkTempFileForWriter(int size, int share_id, char * name)
+{
+	char *tmp_file;
+	int lock;
+
+	tmp_file = (char *)palloc(sizeof(char) * MAXPGPATH);
+	mk_tmp_file_name(tmp_file, size, share_id, name);
+	elog(DEBUG3, "tmp file for writer is %s", tmp_file);
+	fd_tmp_writer = open(tmp_file, O_CREAT, S_IRWXU);
```

Comment:
I do not think we need to provide +wx permission.

I think if the file exists we should quit, right? If yes, we need to add O_EXCL